meta_merkle_tree/src/generated_merkle_tree.rs

```diff
@@ -101,92 +110,130 @@ impl GeneratedMerkleTreeCollection {
 
 #[derive(Clone, Eq, Debug, Hash, PartialEq, Deserialize, Serialize)]
 pub struct TreeNode {
-    /// The stake account entitled to redeem.
     #[serde(with = "pubkey_string_conversion")]
     pub claimant: Pubkey,
-
-    /// Pubkey of the ClaimStatus PDA account, this account should be closed to reclaim rent.
     #[serde(with = "pubkey_string_conversion")]
     pub claim_status_pubkey: Pubkey,
-
-    /// Bump of the ClaimStatus PDA account
     pub claim_status_bump: u8,
-
     #[serde(with = "pubkey_string_conversion")]
     pub staker_pubkey: Pubkey,
-
     #[serde(with = "pubkey_string_conversion")]
     pub withdrawer_pubkey: Pubkey,
-
-    /// The amount this account is entitled to.
     pub amount: u64,
-
-    /// The proof associated with this TreeNode
     pub proof: Option<Vec<[u8; 32]>>,
 }
+
 impl TreeNode {
     fn vec_from_stake_meta(
         stake_meta: &StakeMeta,
-    ) -> Result<Option<Vec<Self>>, MerkleRootGeneratorError> {
+        protocol_fee_bps: u16,
+        tip_distribution_program_id: &Pubkey,
+    ) -> Result<Option<Vec<TreeNode>>, MerkleRootGeneratorError> {
```

Comment:
Why not use `Self` instead of `TreeNode` for return value (and a bunch of places below)?